File: MEMO.DAL/Entities/Attribute.cs
using MEMO.DAL.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace MEMO.DAL.Entities
{
    public class Attribute
    {
        public Guid Id { get; set; }
        [MaxLength(15)]
        public string Name { get; set; }
        public AttributeType Type { get; set; }

        [NotMapped]
        public int AttributeValuesCount {get; set;}

        public Guid UserId { get; set; }
        public User User { get; set; }

        public ICollection<AttributeValue> AttributeValues { get; set; } = new List<AttributeValue>();
        public ICollection<AttributeParameter> AttributeParameters { get; set; } = new List<AttributeParameter>();
    }
}

File: MEMO.MobilClient/app/src/main/java/hu/mobilclient/memo/model/memoapi/Attribute.kt
package hu.mobilclient.memo.model.memoapi

import com.google.gson.annotations.SerializedName
import hu.mobilclient.memo.helpers.Constants
import hu.mobilclient.memo.model.enums.AttributeType
import java.util.*

class Attribute(@SerializedName("id")var Id: UUID = UUID(0,0),
                @SerializedName("user")var User: User = User(),
                @SerializedName("name") var Name: String = Constants.EMPTY_STRING,
                @SerializedName("type") var Type: AttributeType = AttributeType.TEXT,
                @SerializedName("attributeValuesCount")var AttributeValuesCount: Int = Constants.ZERO,
                @SerializedName("attributeParameters") var AttributeParameters: MutableList<AttributeParameter> = ArrayList()){
    override fun toString(): String = Name
}